trendmicro_apex/icon_trendmicro_apex/actions/add_file_to_udso_list/action.py

```diff
@@ -49,6 +47,5 @@ def run(self, params={}):
                 self.logger.error(f"Received status code: {response.status_code}")
                 self.logger.error(f"Response was: {response.text}")
             raise PluginException(assistance="Please verify the connection details and input data.",
```

Comment:
Change this error handling to isolate response.raise_for_status()

The code is this: 
```
        response = None
        try:
            response = requests.put(request_url, headers=self.connection.header_dict, data=json_payload, verify=False)
            response.raise_for_status()
            return {Output.SUCCESS: response is not None}
        except RequestException as rex:
            if response:
                self.logger.error(f"Received status code: {response.status_code}")
                self.logger.error(f"Response was: {response.text}")
            raise PluginException(assistance="Please verify the connection details and input data.",
                                  cause="Error processing the Apex request.",
                                  data=rex)
```

Instead, do this 
```
        response = None
       
       response = requests.put(request_url, headers=self.connection.header_dict, data=json_payload, verify=False)
       try:
            response.raise_for_status()
       except RequestException:
            self.logger.error(f"Received status code: {response.status_code}")
            self.logger.error(f"Response was: {response.text}")
            raise PluginException(assistance="Please verify the connection details and input data.",
                                  cause="Error processing the Apex request.",
                                  data=response.text)

       return {Output.SUCCESS: response is not None}
```

Reply:
Some more notes on this `response = None` is not needed. 

Can we add context around the status code?
ie a 401 and a 404 are not the same thing. dose trend micro document its status codes? 
if they do you can create custom messaging around them. Here is a good example of how to do that.
https://github.com/rapid7/insightconnect-plugins/blob/master/rapid7_insightidr/komand_rapid7_insightidr/util/resource_helper.py

Finally SUPPER picky here but I like to see the plugin exception be `cause, assistance, data` as that is the order in which the customer will see the output

Reply:
We don't change it because in jira there is: "Leave code file the same.". Please add this comments as new ticket and we will fix this :)